packages/media-query/src/create-media-query.ts

```diff
@@ -1,12 +1,17 @@
-import { objectKeys, Dict, isNumber } from "@chakra-ui/utils"
+import { Dict, isNumber, breakpoints as bps } from "@chakra-ui/utils"
 import calculateMeasurement from "calculate-measurement"
 
 /**
  * Create a media query string from the breakpoints,
  * using a combination of `min-width` and `max-width`.
  */
 function createMediaQueries(breakpoints: Dict) {
-  const sorted = getSortedKeys(breakpoints)
+  // get the non-number breakpoint keys from the provided breakpoints
+  const keys = Object.keys(breakpoints)
+
+  // use only the keys matching the official breakpoints names, and sort them in
+  // largest to smallest order
+  const sorted = bps.filter((bp) => keys.includes(bp)).reverse()
```

Comment:
Instead of using all of the keys that are on `breakpoints` args (which includes stringified array indexes since it'll be an array with keys attached to its prototype), we instead use the `breakpoints` array from `@chakra-ui/utils` and only include keys that are present in the object passed to the function. 

This guarantees that we're only using valid breakpoints, and that we're using the actual breakpoint names.

We also reverse them to make sure we're returning an array of largest to smallest breakpoints.